zebrad/src/commands/start.rs

```diff
@@ -601,3 +606,43 @@ impl config::Override<ZebradConfig> for StartCmd {
         Ok(config)
     }
 }
+
+/// Check if there are old database folders and delete them from the filesystem.
+///
+/// Iterate over the files and directories in the databases folder and delete if:
+/// - The entry is a directory.
+/// - The directory name has a lenght of at least 2 characters.
+/// - The directory name has a prefix `v`.
+/// - The directory name without the prefix can be parsed as an unsigned number.
+/// - The parsed number is lower than the hardcoded `DATABASE_FORMAT_VERSION`.
+fn check_and_delete_old_databases(cache_dir: PathBuf) -> Result<(), Report> {
```

Comment:
I was thinking on putting this somewhere else but not sure where so ended up here.